### PR TITLE
telemeter-server: Always set OpenShift telemetry tenant ID

### PR DESCRIPTION
**Pull request by brancz**
@metalmatze @squat @kakkoyun @bwplotka @krasi-georgiev

### CODE REVIEW

File: pkg/receive/handler.go [hunk 1]
```diff
@@ -79,7 +77,7 @@ func (h *Handler) Receive(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	req = req.WithContext(ctx)
-	req.Header.Add("THANOS-TENANT", r.Context().Value(authorize.TenantKey).(string))
```
Comment (squat) [inline pkg/receive/handler.go:82]: maybe we should make this configurable via a flag so we don't have to make a new PR to telemeter every time the tenant ID changes. WDYT?

Comment (brancz) [reply pkg/receive/handler.go:82]: why would it change? it's kind of final ..

Comment (squat) [reply pkg/receive/handler.go:82]: when we run in different environments, or someone else runs this somewhere

Comment (metalmatze) [reply pkg/receive/handler.go:82]: Hopefully, this is replaced by OIDC soo anyway, no? If so, we could just leave it as is and then update to the proper new flow later.

**Comment by brancz** [comment]
Merging this as we need it for our production system. This has nothing to do with anything shipped in OpenShift, it's entirely server side.

**Pull request merged**